packages/compiler_utils/src/file.candy

```diff
@@ -25,6 +25,16 @@ public class FancyFile {
       Tuple(File(getResolvedPath(context)).read(), List.empty<CompilerError>())
     })
   }
+
+  // TODO(marcelgarus): This is quite ugly. We should rename this class to `FancyPath` and then
+  // create `FancyFile` and `FancyDirectory` wrappers that offer additional functionality.
+  fun children(context: QueryContext<List<CompilerError>>): List<FancyFile> {
+    let packagePath = getResolvedPath(context)
+    (Directory(getResolvedPath(context))
+      .listContents(false) as Iterable<FileSystemNode>)
+      .map<FancyFile>({ FancyFile(package, it.path.relativeTo(packagePath)) })
+      .toList()
```

Comment:
Shouldn't this be an impure query? Otherwise, we might not detect new/deleted files when incremental compilation works.